metadata/src/main/java/org/apache/kafka/controller/PartitionChangeBuilder.java

```diff
@@ -405,7 +415,7 @@ private void completeReassignmentIfNeeded() {
         targetAdding = Collections.emptyList();
     }
 
-    public Optional<ApiMessageAndVersion> build() {
+    public Optional<ApiMessageAndVersion> build(DefaultDirProvider defaultDirProvider) {
```

Comment:
Please don't add arguments to build(). If you want to have a way to set the default directory, then have a `setDefaultDirProvider` function.

Reply:
Updated this.